maptivist/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -107,91 +81,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
         }
         return markers;
     }
  
-    public String createCategoriesString(Set<String> categories){
-        String categoryString = "";
-        Base64.Encoder encoder = Base64.getEncoder();  
-        for (String category : categories) {
-            String categoryByte = encoder.encodeToString(category.getBytes());  
-            categoryString += categoryByte + ",";
-        }
-        return categoryString.substring(0, categoryString.length() - 1);     
-    }
- 
-    public Set<String> createCategoriesObject(String categoryString) {
-        Base64.Decoder decoder = Base64.getDecoder();  
-        Set<String> categorySet = new HashSet<String>();
-        String[] categories;
-        categories = categoryString.split(",");
-        for (String category : categories) {
-            String categoryDecode = new String(decoder.decode(category));
-            categorySet.add(categoryDecode);
-        }
-        return categorySet;
-    }
- 
-    public String createFlagString(ArrayList<String> flags) {
-        String flagString = "";
-        Base64.Encoder encoder = Base64.getEncoder();  
-        for (String flag : flags) {
-            String comment = encoder.encodeToString(flag.getBytes());  
-            flagString += comment + ",";
-        }
-        return flagString.substring(0, flagString.length() - 1);
-    }
- 
-    public ArrayList<String> createFlagObject(String flagString) {
-        Base64.Decoder decoder = Base64.getDecoder();  
-        ArrayList<String> flagsList = new ArrayList<String>();
-        String[] flags;
-        flags = flagString.split(",");
-        for (String flag : flags) {
-            String flagDecode = new String(decoder.decode(flag));
-            flagsList.add(flagDecode);
-        }
-        return flagsList;
-    }
- 
-    public String createLinkString(Set<String> links) {
-        String linkString = "";
-        Base64.Encoder encoder = Base64.getUrlEncoder();  
-        for (String link : links) {
-            String linkUrl = encoder.encodeToString(link.getBytes());  
-            linkString += linkUrl + ",";
-        }
-        return linkString.substring(0, linkString.length() - 1);
-    }
- 
-    public Set<String> createLinkObject(String linkString) {
-        Base64.Decoder decoder = Base64.getDecoder();  
-        Set<String> linkList = new HashSet<String>();
-        String[] links;
-        links = linkString.split(",");
-        for (String link : links) {
-            String linkDecode = new String(decoder.decode(link));
-            linkList.add(linkDecode);
-        }
-        return linkList;
-    }
-
-    public Boolean checkIfMarkerAlreadyInDatastore(String title){
+    public Boolean checkIfMarkerAlreadyInDatastore(UUID id){
+        String id_string = id.toString();
         Query query = new Query("Marker");
-        query.addFilter("title", Query.FilterOperator.EQUAL, title);
+        query.addFilter("id", Query.FilterOperator.EQUAL, id_string);
```

Comment:
You may consider also changing getEntity to take the UUID instead of the title, as well.